Sources/PackageModel/Module.swift

```diff
@@ -129,9 +130,24 @@ public class ClangModule: CModule {
     }
 }
 
+/// A dictionary containing mapping of C family source extension to their file type in Xcode.
+private let xcodeCLanguageFileTypeDict = [
+    ".c": "sourcecode.c.c",
+    ".m": "sourcecode.c.objc",
+    ".cxx": "sourcecode.cpp.cpp",
+    ".cc": "sourcecode.cpp.cpp",
+    ".cpp": "sourcecode.cpp.cpp",
+    ".mm": "sourcecode.cpp.objcpp",
+]
+
 extension ClangModule: XcodeModuleProtocol {
-    public var fileType: String {
-        return "sourcecode.c.c"
+    public func fileType(forSource source: String) -> String {
```

Comment:
but you also do `source.hasSuffix(key)` check.
source.hasSuffix(key)  would be equal to `source.fileExtension` and xcodeCLanguageFileTypeDict[fileExtension] is perforce vise better than iterating over dic.

Also I feel it's more clean intention if you get file extension and get its value from dictionary